caricazione delle librerie, del df e modifica del df per dividere gli stati in continenti


In [3]:
# Manipolazione dati
import pandas as pd
import numpy as np

# Visualizzazione
import matplotlib.pyplot as plt
import seaborn as sns

# Statistica
from scipy import stats
from scipy.stats import pearsonr, spearmanr

df = pd.read_csv('cause_of_deaths.csv')



# ==================== AGGIUNGERE COLONNA CONTINENT ====================

# Dizionario che mappa ogni paese al suo continente
country_to_continent = {
    # AFRICA
    'Afghanistan': 'Asia', 'Algeria': 'Africa', 'Angola': 'Africa', 'Benin': 'Africa',
    'Botswana': 'Africa', 'Burkina Faso': 'Africa', 'Burundi': 'Africa', 'Cameroon': 'Africa',
    'Cape Verde': 'Africa', 'Central African Republic': 'Africa', 'Chad': 'Africa',
    'Comoros': 'Africa', 'Congo': 'Africa', "Cote d'Ivoire": 'Africa', 
    'Democratic Republic of Congo': 'Africa', 'Djibouti': 'Africa', 'Egypt': 'Africa',
    'Equatorial Guinea': 'Africa', 'Eritrea': 'Africa', 'Ethiopia': 'Africa', 'Gabon': 'Africa',
    'Gambia': 'Africa', 'Ghana': 'Africa', 'Guinea': 'Africa', 'Guinea-Bissau': 'Africa',
    'Kenya': 'Africa', 'Lesotho': 'Africa', 'Liberia': 'Africa', 'Libya': 'Africa',
    'Madagascar': 'Africa', 'Malawi': 'Africa', 'Mali': 'Africa', 'Mauritania': 'Africa',
    'Mauritius': 'Africa', 'Morocco': 'Africa', 'Mozambique': 'Africa', 'Namibia': 'Africa',
    'Niger': 'Africa', 'Nigeria': 'Africa', 'Rwanda': 'Africa', 'Sao Tome and Principe': 'Africa',
    'Senegal': 'Africa', 'Seychelles': 'Africa', 'Sierra Leone': 'Africa', 'Somalia': 'Africa',
    'South Africa': 'Africa', 'South Sudan': 'Africa', 'Sudan': 'Africa', 'Swaziland': 'Africa',
    'Tanzania': 'Africa', 'Togo': 'Africa', 'Tunisia': 'Africa', 'Uganda': 'Africa',
    'Zambia': 'Africa', 'Zimbabwe': 'Africa',
    
    # ASIA
    'Armenia': 'Asia', 'Azerbaijan': 'Asia', 'Bahrain': 'Asia', 'Bangladesh': 'Asia',
    'Bhutan': 'Asia', 'Brunei': 'Asia', 'Cambodia': 'Asia', 'China': 'Asia',
    'Georgia': 'Asia', 'India': 'Asia', 'Indonesia': 'Asia', 'Iran': 'Asia',
    'Iraq': 'Asia', 'Israel': 'Asia', 'Japan': 'Asia', 'Jordan': 'Asia',
    'Kazakhstan': 'Asia', 'Kuwait': 'Asia', 'Kyrgyzstan': 'Asia', 'Laos': 'Asia',
    'Lebanon': 'Asia', 'Malaysia': 'Asia', 'Maldives': 'Asia', 'Mongolia': 'Asia',
    'Myanmar': 'Asia', 'Nepal': 'Asia', 'North Korea': 'Asia', 'Oman': 'Asia',
    'Pakistan': 'Asia', 'Palestine': 'Asia', 'Philippines': 'Asia', 'Qatar': 'Asia',
    'Saudi Arabia': 'Asia', 'Singapore': 'Asia', 'South Korea': 'Asia', 'Sri Lanka': 'Asia',
    'Syria': 'Asia', 'Taiwan': 'Asia', 'Tajikistan': 'Asia', 'Thailand': 'Asia',
    'Timor': 'Asia', 'Turkey': 'Asia', 'Turkmenistan': 'Asia', 'United Arab Emirates': 'Asia',
    'Uzbekistan': 'Asia', 'Vietnam': 'Asia', 'Yemen': 'Asia',
    
    # EUROPE
    'Albania': 'Europe', 'Andorra': 'Europe', 'Austria': 'Europe', 'Belarus': 'Europe',
    'Belgium': 'Europe', 'Bosnia and Herzegovina': 'Europe', 'Bulgaria': 'Europe',
    'Croatia': 'Europe', 'Cyprus': 'Europe', 'Czech Republic': 'Europe', 'Czechia': 'Europe',
    'Denmark': 'Europe', 'Estonia': 'Europe', 'Finland': 'Europe', 'France': 'Europe',
    'Germany': 'Europe', 'Greece': 'Europe', 'Hungary': 'Europe', 'Iceland': 'Europe',
    'Ireland': 'Europe', 'Italy': 'Europe', 'Kosovo': 'Europe', 'Latvia': 'Europe',
    'Liechtenstein': 'Europe', 'Lithuania': 'Europe', 'Luxembourg': 'Europe',
    'Macedonia': 'Europe', 'Malta': 'Europe', 'Moldova': 'Europe', 'Monaco': 'Europe',
    'Montenegro': 'Europe', 'Netherlands': 'Europe', 'Norway': 'Europe', 'Poland': 'Europe',
    'Portugal': 'Europe', 'Romania': 'Europe', 'Russia': 'Europe', 'San Marino': 'Europe',
    'Serbia': 'Europe', 'Slovakia': 'Europe', 'Slovenia': 'Europe', 'Spain': 'Europe',
    'Sweden': 'Europe', 'Switzerland': 'Europe', 'Ukraine': 'Europe', 'United Kingdom': 'Europe',
    'Vatican': 'Europe', 'North Macedonia': 'Europe',
    
    # NORTH AMERICA
    'Antigua and Barbuda': 'North America', 'Bahamas': 'North America', 'Barbados': 'North America',
    'Belize': 'North America', 'Canada': 'North America', 'Costa Rica': 'North America',
    'Cuba': 'North America', 'Dominica': 'North America', 'Dominican Republic': 'North America',
    'El Salvador': 'North America', 'Grenada': 'North America', 'Guatemala': 'North America',
    'Haiti': 'North America', 'Honduras': 'North America', 'Jamaica': 'North America',
    'Mexico': 'North America', 'Nicaragua': 'North America', 'Panama': 'North America',
    'Saint Kitts and Nevis': 'North America', 'Saint Lucia': 'North America',
    'Saint Vincent and the Grenadines': 'North America', 'Trinidad and Tobago': 'North America',
    'United States': 'North America', 'Guam': 'North America', 'Puerto Rico': 'North Ameica',
    'United States Virgin Islands': 'North America',
    
    # SOUTH AMERICA
    'Argentina': 'South America', 'Bolivia': 'South America', 'Brazil': 'South America',
    'Chile': 'South America', 'Colombia': 'South America', 'Ecuador': 'South America',
    'Guyana': 'South America', 'Paraguay': 'South America', 'Peru': 'South America',
    'Suriname': 'South America', 'Uruguay': 'South America', 'Venezuela': 'South America',
    
    # OCEANIA
    'Australia': 'Oceania', 'Fiji': 'Oceania', 'Kiribati': 'Oceania', 'Marshall Islands': 'Oceania',
    'Micronesia': 'Oceania', 'Nauru': 'Oceania', 'New Zealand': 'Oceania', 'Palau': 'Oceania',
    'Papua New Guinea': 'Oceania', 'Samoa': 'Oceania', 'Solomon Islands': 'Oceania',
    'Tonga': 'Oceania', 'Tuvalu': 'Oceania', 'Vanuatu': 'Oceania', 'American Samoa': "Oceania", 
    'Bermuda': "Oceania", 'Cook Islands': "Oceania",  'Eswatini': "Africa", 'Greenland': "North America",
    'Niue': 'Oceania', 'Northern Mariana Islands': 'Oceania', 'Tokelau': 'Oceania',
}

# Aggiungi la colonna Continent
df['Continent'] = df['Country/Territory'].map(country_to_continent)


# ===== INFO GENERALI =====
print("="*60)
print("INFORMAZIONI DATASET")
print("="*60)
print(f"Dimensioni: {df.shape}")
print(f"Periodo: {df['Year'].min()} - {df['Year'].max()}")
print(f"Numero paesi: {df['Country/Territory'].nunique()}")
print(f"Numero continenti: {df['Continent'].nunique()}")

# Info tipi di dato
df.info()




INFORMAZIONI DATASET
Dimensioni: (6120, 35)
Periodo: 1990 - 2019
Numero paesi: 204
Numero continenti: 7
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6120 entries, 0 to 6119
Data columns (total 35 columns):
 #   Column                                      Non-Null Count  Dtype 
---  ------                                      --------------  ----- 
 0   Country/Territory                           6120 non-null   object
 1   Code                                        6120 non-null   object
 2   Year                                        6120 non-null   int64 
 3   Meningitis                                  6120 non-null   int64 
 4   Alzheimer's Disease and Other Dementias     6120 non-null   int64 
 5   Parkinson's Disease                         6120 non-null   int64 
 6   Nutritional Deficiencies                    6120 non-null   int64 
 7   Malaria                                     6120 non-null   int64 
 8   Drowning                                    6120 non-null   int6

Il dataset "Global Causes of Death" contiene dati sulla mortalità 
per 33 diverse cause in 204 paesi dal 1990 al 2019.
I dati provengono da [fonte Kaggle/OMS] e rappresentano il numero 
assoluto di morti per anno.
Analizzeremo l'evoluzione temporale e i pattern geografici delle 
principali cause di morte.
Abbiamo identificato 3 Domande di Ricerca che ci consentiranno di usare i dati diversi per ogni domanda.

### Q1 (Temporale): Le morti per Alzheimer sono aumentate significativamente nel periodo 1990-2019?
### Q2 (Geografica): Esistono differenze significative nelle morti per malattie cardiovascolari tra continenti?
### Q3 (Correlazione): Esiste una correlazione negativa tra morti per Alzheimer e morti per Malaria?

Ma prima controlliamo se ci sono valori mancanti:


In [6]:
#ANALISI VALORI MANCANTI

print("Valori mancanti:")

mancanti = df.isnull().sum()
percentuale_mancanti = (mancanti / len(df)) * 100

df_mancanti = pd.DataFrame({
    "Conteggio mancanti": mancanti,
    "Percentuale": percentuale_mancanti
})


df_mancanti = df_mancanti[df_mancanti["Conteggio mancanti"] > 0].sort_values("Conteggio mancanti", ascending= False)
if len(df_mancanti) == 0:
    print("✅ Nessun valore mancante!")
else:
    print(df_mancanti)


Valori mancanti:
✅ Nessun valore mancante!


#### Perfetto! Non sono presenti valori mancanti

Ora ci conviene mettere in focus le variabili che ci saranno più utili: Country/Territory, Continent, Year, Alzheimer's Disease and Other Dementias,Malaria, Cardiovascuar Diseases


In [ ]:
variabili_focus = ['Country/Territory', 'Continent', 'Year', 
              'Alzheimer\'s Disease and Other Dementias',
              'Malaria', 
              'Cardiovascular Diseases']

df_focus = df[variabili_focus].copy()


#rinominiamole per comodità
df_focus.columns = ['Country', 'Continent', 'Year', 'Alzheimer', 'Malaria', 'Cardiovascular']


print(df_focus)


          Country Continent  Year  Alzheimer  Malaria  Cardiovascular
0     Afghanistan      Asia  1990       1116       93           44899
1     Afghanistan      Asia  1991       1136      189           45492
2     Afghanistan      Asia  1992       1162      239           46557
3     Afghanistan      Asia  1993       1187      108           47951
4     Afghanistan      Asia  1994       1211      211           49308
...           ...       ...   ...        ...      ...             ...
6115     Zimbabwe    Africa  2015        754     2518           16649
6116     Zimbabwe    Africa  2016        767     2050           16937
6117     Zimbabwe    Africa  2017        781     2116           17187
6118     Zimbabwe    Africa  2018        795     2088           17460
6119     Zimbabwe    Africa  2019        812     2068           17810

[6120 rows x 6 columns]


### Analisi degli Outlier
Gli outlier sono valori anomali che si discostano significativamente dalla distribuzione generale dei dati, influenzando negativamente la stabilità delle analisi statistiche.

Identificheremo gli outliers tramite dei boxplot.

In [ ]:
fig, axes = plt.subplots(1, 3, figsize=(15, 5))

axes[0].boxplot(df_focus['Alzheimer'].dropna())
axes[0].set_title('Alzheimer - Boxplot')
axes[0].set_ylabel('Deaths')

axes[1].boxplot(df_focus['Malaria'].dropna())
axes[1].set_title('Malaria - Boxplot')
axes[1].set_ylabel('Deaths')

axes[2].boxplot(df_focus['Cardiovascular'].dropna())
axes[2].set_title('Cardiovascular - Boxplot')
axes[2].set_ylabel('Deaths')

plt.tight_layout()
plt.show()

# Top 5 paesi per ogni malattia
print("\n📊 Top 5 paesi - Alzheimer (2019):")
print(df_focus[df_focus['Year']==2019].nlargest(5, 'Alzheimer')[['Country', 'Alzheimer']])

print("\n📊 Top 5 paesi - Malaria (2019):")
print(df_focus[df_focus['Year']==2019].nlargest(5, 'Malaria')[['Country', 'Malaria']])

print("\n📊 Top 5 paesi - Cardiovascular (2019):")
print(df_focus[df_focus['Year']==2019].nlargest(5, 'Cardiovascular')[['Country', 'Cardiovascular']])
